test/metabase/server/statistics_handler_test.clj

```diff
@@ -0,0 +1,252 @@
+(ns metabase.server.statistics-handler-test
+  (:require
+   [clojure.core.async :as a]
+   [clojure.test :refer [deftest testing is]]
+   [metabase.analytics.prometheus-test :as prometheus-test]
+   [metabase.server.statistics-handler :as sut]
+   [metabase.test :as mt])
+  (:import
+   (jakarta.servlet AsyncListener)
+   (jakarta.servlet.http HttpServletRequest HttpServletResponse)
+   (org.eclipse.jetty.ee9.nested HandlerWrapper Request)
+   (org.eclipse.jetty.ee9.servlet ServletHandler ServletContextHandler)
+   (org.eclipse.jetty.server LocalConnector Server)))
+
+(set! *warn-on-reflection* true)
+
+(defn on-complete-listener
+  [done-channel]
+  (proxy [AsyncListener] []
+    (onComplete [_]
+      (a/>!! done-channel :done))))
+
+(defn- async-servlet-handler
+  ^ServletHandler [status-code chan-start-handle chan-finish-handle]
+  (proxy [ServletHandler] []
+    (doHandle [_ ^Request request _ ^HttpServletResponse response]
+      (let [async-context (.startAsync request)]
+        (.setHandled request true)
+        (a/<!! chan-start-handle)
+        (a/go
+          (.setStatus response status-code)
+          (.setContentLength response 0)
+          (.. response getOutputStream close)
+          (a/<!! chan-finish-handle)
```

Comment:
why does the async servlet handler read from the `chan-finish-handle` twice?

Reply:
because I need it to block twice to coordinate the tests at the bottom. I think this could be rewritten (maybe in the first on master) to just have one coordination channel instead of the start/finish ones which make sense in the sync case but are confusing in the async one.